chapter_appendix/information_theory.md

```diff
@@ -0,0 +1,371 @@
+# Information Theory
+:label:`chapter_information_theory`
+
+Machine learning, which studies how computer systems can use data to improve performance, is aimed to extract critical and interesting signals from data. While, information theory, which studies encoding, decoding, transmitting and manipulating information (aka data), can help us measure and compare how much information are presented in different signals. As a result, information theory provides a strong backbone to the fundamental of machine learning. For example, “cross entropy loss” is used as an objective function in many machine learning algorithms. This is a direct application of an information theory to measure the loss. In addition, in the rule-based learning algorithm (such as the decision tree algorithm), we can use entropy as a unit to quantify the amount of information gained from the best split at each node. In this section, we will talk about the most fundamental concept of information theory and its applications via several examples.
+
+## Information
+### Motivation
+
+"Information" is conveyed or represented by a particular arrangement or sequence of things. It cares about the abstract possibility rather than the "knowledge" within the entity. Suppose there is a file on your laptop and you are curious about how much information it contains. No matter whether it is a song, a movie or a text document, you may find the amount of information is measured in a unit of *bit*, for the size of the file. 
+
+So what is a "bit" and why do we use it to measure information? Historically, an antique transmitter can only send or receive two kinds of code: "0" and "1". In this way, any information symbol is encoded by a series of "0" and "1" combinations, and each digit represents 1 *bit*. 
+
+But since information is intereted in the "abstract possibility", how do we map the possibility to the number of bits? Recall that for any series of codes, each "0" or "1" occurs with a probability of $\frac{1}{2}$. Hence, a series of codes with length $n$ occurs with a probability of $\frac{1}{2^n}$. At the same times, this series contains $n$ bits of information. Therefore, we can define *self-information (or surprisal)*: 
+
```

Comment:
"But since" -> "However, since..."

"Recall that" -> Suppose that ..

At the same time,

delete "(or surprisal)"

Reply:
no ":" in the end

Reply:
"surprisal" needs more context